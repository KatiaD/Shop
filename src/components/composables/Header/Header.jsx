import React from "react";
import { Link } from "react-router-dom";
import { HeaderWrapper, HeaderTitle } from "./Header.styled";

const displayName = "Header";

const defaultProps = {};

function Header({ title }) {
  return (
    <HeaderWrapper>
      <HeaderTitle>
        Sports Store - {title}
        {title == "Products" && <Link to="/cart"> Cart</Link>}
      </HeaderTitle>
    </HeaderWrapper>
  );
}

Header.displayName = displayName;
Header.defaultProps = defaultProps;

export default Header;
