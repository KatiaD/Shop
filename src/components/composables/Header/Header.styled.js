import styled from "styled-components";

export const HeaderWrapper = styled.header`
  display: flex;
  justify-content: center;
  align-content: space-between;
  align-items: center;
  flex-direction: column;
  padding: 20px;
  background-color: #ffa50054;
`;

export const HeaderTitle = styled.h1`
  font-size: 20px;
`;
