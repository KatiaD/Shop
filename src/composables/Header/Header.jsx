import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import cart from 'images/cart.png';
import {
  HeaderWrapper, HeaderTitle, Cart, CartWrapper, Total,
} from './Header.styled';

const displayName = 'Header';

const propTypes = {
  title: PropTypes.string,
  total: PropTypes.number,
};

const defaultProps = {
  title: '',
  total: 0,
};

function Header({ title, total }) {
  return (
    <HeaderWrapper>
      <HeaderTitle>
        Watches Store -
        {title}
        <CartWrapper>
          {title === 'Products' && (
            <Link to="/cart" data-name="header-cart">
              <Cart src={cart} alt="" />
              <Total>{total}</Total>
            </Link>
          )}
        </CartWrapper>
      </HeaderTitle>
    </HeaderWrapper>
  );
}

Header.displayName = displayName;
Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
