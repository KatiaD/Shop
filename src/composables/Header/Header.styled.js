import styled from 'styled-components';

export const HeaderWrapper = styled.header`
  flex: 0 0 auto;
  background-color: black;
  padding-left: 80px;
`;

export const HeaderTitle = styled.h1`
  font-size: 14px;
  color: white;
  font-family: arial;
  font-weight: normal;
  text-transform: uppercase;
`;

export const Cart = styled.img`
  width: 14px;
`;

export const CartWrapper = styled.div`
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  top: 10px;
  right: 20px;
  width: 60px;
`;

export const Total = styled.div`
  color: white;
  float: right;
  font-size: 10px;
`;