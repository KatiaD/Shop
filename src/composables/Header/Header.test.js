import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Given the Header component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeEach(() => {
      component = shallow(<Header/>);
    });

    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});