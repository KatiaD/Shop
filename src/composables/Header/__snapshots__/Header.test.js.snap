// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Header component when the component is rendered should render the component with provided props 1`] = `
<Headerstyled__HeaderWrapper>
  <Headerstyled__HeaderTitle>
    Watches Store - 
    <Headerstyled__CartWrapper />
  </Headerstyled__HeaderTitle>
</Headerstyled__HeaderWrapper>
`;
